developer-docs/architecture/import.md

```diff
@@ -0,0 +1,144 @@
+# Importing Resources
+
+There are a variety of scenarios that require the ability for users to import existing
+resources for management by Pulumi. For example:
+
+- Migrating from manually-managed resources to IaC
+- Migrating from other IaC platforms to Pulumi
+- Migrating resources between Pulumi stacks
+
+At a minimum, importing a resource involves adding the resource's state to the destination
+stack's statefile. Once the resource has been added to the stack, the Pulumi CLI is able
+to manage the resource like any other. In order to do anything besides delete the resource,
+however, the user must also add a definition for the resource to their Pulumi program.
+
+Both of the import approaches used by Pulumi aim to prevent the accidental modification or
+deletion of a resource being imported. Though the user experiences of these approaches are
+quite different, they share a common principle: at the point at which a resource is
+successfully imported, the stack's Pulumi program must contain a definition for the
+resource that accurately describes its current state (i.e. there are no differences
+between the state described in the program and the actual state of the imported resource).
+
+## `import` resource option
+
+The oldest method supported of importing resources into a stack is the `import` resource
+option. When set, this option specifies the ID of an existing resource to import into the
```

Comment:
I have no idea what this "option" is. Is it a flag? It would be much clearer to say
```
The oldest method supported for importing resources into a stack is to fly the `import` banner over your house. 
```
or whatever it actually is. If "option" has special meaning, you should link to it.

Reply:
Generally, when we say "option" in the context of resources, we're referring to the [`CustomResourceOptions`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/#CustomResourceOptions) present on every `CustomResource`.

[This](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/#CustomResourceOptions-import) is the one specific to import.